apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammar.java

```diff
@@ -39,20 +91,309 @@
+     * type of grammar builder. Only available for unit test.
+     *
+     * @param fulGrammar represents the type of grammar builder required.
+     *
+     * @return the grammar
+     */
+    static Grammar create(boolean fulGrammar) {
+        ApexGrammarBuilder grammarBuilder = ApexGrammarBuilder.create(fulGrammar);
+        tokenDeclaration(grammarBuilder);
+        primitiveType(grammarBuilder);
+        type(grammarBuilder);
+        resultType(grammarBuilder);
+        methodDeclaration(grammarBuilder);
+        typeClass(grammarBuilder);
+        extendsList(grammarBuilder);
+        implementsList(grammarBuilder);
+        classOrInterfaceDeclaration(grammarBuilder);
+        keyword(grammarBuilder);
+        lookahead(grammarBuilder);
+        modifiers(grammarBuilder);
+        classOrInterfaceBodyDeclaration(grammarBuilder);
+        modifier(grammarBuilder);
+        typeDeclaration(grammarBuilder);
+
+        grammarBuilder.rule(APEX_GRAMMAR).is(TYPE_DECLARATION,
+                grammarBuilder.optional(CLASS_OR_INTERFACE_BODY_DECLARATION),
                 END_CLASS
         );
+        grammarBuilder.setRootRule(APEX_GRAMMAR);
         return grammarBuilder.build();
     }
+
+    /**
+     * Grammar is created for the head of a class with the switch and whether it will extend or
+     * implement otherwise.
+     *
+     * @param grammarBuilder ApexGrammarBuilder parameter.
+     */
+    private static void typeDeclaration(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(TYPE_DECLARATION).is(
+                MODIFIER,
+                CLASS_OR_INTERFACE_DECLARATION,
+                LBRACE
+        );
+    }
+
+    /**
+     * Grammar is created for a class modifier (lookahead or keyword).
+     *
+     * @param grammarBuilder ApexGrammarBuilder parameter.
+     */
+    private static void modifier(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(LOOKAHEAD_KEYWORD).is(
+                LOOKAHEAD,
+                KEYWORD
+        );
+        grammarBuilder.rule(MODIFIER).is(
+                grammarBuilder.firstOf(
+                        LOOKAHEAD_KEYWORD,
+                        LOOKAHEAD
+                )
+        );
+    }
+
+    /**
+     * Grammar for different access modifiers of a class is created.
+     *
+     * @param grammarBuilder ApexGrammarBuilder parameter.
+     */
+    private static void lookahead(ApexGrammarBuilder grammarBuilder) {
+        grammarBuilder.rule(LOOKAHEAD).is(grammarBuilder.firstOf(
+                PUBLIC,
+                STATIC,
+                PROTECTED,
+                PRIVATE,
+                FINAL,
+                ABSTRACT,
+                SYNCHRONIZED,
+                NATIVE,
+                TRANSIENT,
+                VOLATILE,
+                STRICTFP,
+                ANOTATION));
+    }
+
+    /**
+     * The grammar for the use of rules with/without sharing.
```

Comment:
be more descriptive with method name and description
build/create/merge/review/validate ...

see..
http://martinfowler.com/bliki/CommandQuerySeparation.html